File: study_abroad/HenCoder/APT/BindView/src/main/java/com/xuebinduan/bindview/MainActivity.java
package com.xuebinduan.bindview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.xuebinduan.lib.Binding;
import com.xuebinduan.lib_annotations.BindView;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.text_view)
    public TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        textView = findViewById(R.id.text_view);
        Binding.bind(this);

        textView.setText("Hi,YOU!");

    }
}

File: view_binding/src/main/java/com/xuebinduan/view_binding/MainActivity.kt
package com.xuebinduan.view_binding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.xuebinduan.view_binding.databinding.ActivityMainBinding


/**
 * ViewBinding还是要通过setContentView来设置视图；
 * 唯一的方便点就是：我们可以不用findViewById，直接通过binding.id就可以获取到它
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        val contentView = binding.root
        setContentView(contentView)

        binding.textView.text = "This is ViewBinding"
    }
}